src/commands/functions/create.js

```diff
@@ -258,10 +258,7 @@ const downloadFromURL = async function (context, flags, args, functionsDir) {
     const { onComplete, addons = [] } = require(fnTemplateFile)
 
     await installAddons(context, addons, path.resolve(fnFolder))
-    if (onComplete) {
-      await addEnvVariables(context.netlify.api, context.netlify.site)
```

Comment:
This was done so the `onComplete` handler runs in the same context as `dev`, but missing dotenv envs.
This PR fixes that